### PR TITLE
Updated proguard-rules to make sure that mobilesdk classes are kept, …

### LINKED ISSUE #3632: Unable to get implementation class: Didn't find class "org.lantern.mobilesdk.embedded.EmbeddedLantern"
Looks like the new dynamic class loading in ``doEnable`` is causing issues with release builds that's preventing Lantern from proxying. After building a new release off valencia, I encountered this fatal error when I switch Lantern on:

```
VpnService  E  Fatal error
 E  org.b.a.b: Unable to get implementation class: Didn't find class "org.lantern.mobilesdk.embedded.EmbeddedLantern" on path: DexPathList[[zip fil e "/data/app/org.getlantern.lantern-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.getlantern.lantern-1, /vendor/lib, /system/lib]] E      at org.b.a.a.bo(Lantern.java:85)
 E      at org.b.a.a.b(Lantern.java:59)
 E      at org.b.a.a.a(Lantern.java:40)
 E      at org.getlantern.lantern.vpn.Service.run(Service.java:47) E      at java.lang.Thread.run(Thread.java:841)
 E  Caused by: java.lang.ClassNotFoundException: Didn't find class "org.lantern.mobilesdk.embedded.EmbeddedLantern" on path: DexPathList[[zip file "/data/app/org.getlantern.lantern-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.getlantern.lantern-1, /vendor/lib, /system/lib]] E      at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:67) E      at java.lang.ClassLoader.loadClass(ClassLoader.java:497) E      at java.lang.ClassLoader.loadClass(ClassLoader.java:457) E      at org.b.a.a.bo(Lantern.java:82)
 E      ... 4 more
 E  Lantern terminated.
```

### PR DESCRIPTION
**Pull request by oxtoacart**
…closes #3632

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -590,9 +590,8 @@ $(LANTERN_MOBILE_ANDROID_DEBUG): $(LANTERN_MOBILE_TUN2SOCKS) $(LANTERN_MOBILE_AN
 
 $(LANTERN_MOBILE_ANDROID_RELEASE): $(LANTERN_MOBILE_TUN2SOCKS) $(LANTERN_MOBILE_ANDROID_LIB) $(LANTERN_MOBILE_ANDROID_SDK)
 	@echo "Generating distribution package for android..."
-	ln -s $$SECRETS_DIR/android/keystore.release.jks $(LANTERN_MOBILE_DIR)/app
+	ln -f -s $$SECRETS_DIR/android/keystore.release.jks $(LANTERN_MOBILE_DIR)/app
```
Comment (oxtoacart) [inline Makefile]: This step failed after the first time because the symlink already existed.  Adding -f fixed that.

File: Makefile [hunk 2]
```diff
@@ -590,9 +590,8 @@ $(LANTERN_MOBILE_ANDROID_DEBUG): $(LANTERN_MOBILE_TUN2SOCKS) $(LANTERN_MOBILE_AN
...
 	cd $(LANTERN_MOBILE_DIR)/app
-	cp $(ANDROID_SDK_ANDROID_LIB) $(LANTERN_MOBILE_LIBS)
```
Comment (oxtoacart) [inline Makefile]: This step is unnecessary, as the LANTERN_MOBILE_ANDROID_SDK target already does this.

File: MobileSDK/sdk/proguard-rules.pro [hunk 3]
```diff
@@ -15,3 +15,6 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
+
+# Keep
+-keep class org.lantern.mobilesdk.**
```
Comment (oxtoacart) [inline MobileSDK/sdk/proguard-rules.pro]: This is the fix.  I put it in the MobileSDK project and then reference it from the lantern-mobile project.

**Comment by myleshorton** [comment]
LGTM @oxtoacart. Let's just wait for travis.

**Comment by atavism** [comment]
Same here!

**Pull request merged**